### ISSUE TITLE
Reduce computational demands of acquiring unregularized dropout

### ISSUE THREAD

**Issue by Ricu** [Ricu/fantastic-umbrella#1]
Currently, the completed backwards pass is executed when trying to acquire the gradient with respect to the logits. This introduces a lot of unnecessary computations.

- Idea: cut the execution graph after the logits.
